include/pybind11/embed.h

```diff
@@ -261,6 +281,16 @@ class scoped_interpreter {
         initialize_interpreter(init_signal_handlers, argc, argv, add_program_dir_to_path);
     }
 
+#if PY_VERSION_HEX >= PYBIND11_PYCONFIG_SUPPORT_PY_VERSION_HEX
+    explicit scoped_interpreter(PyConfig *config,
+                                int argc = 0,
+                                const char *const *argv = nullptr,
+                                bool add_program_dir_to_path = true) {
+        detail::precheck_interpreter();
+        detail::initialize_interpreter(config, argc, argv, add_program_dir_to_path);
```

Comment:
I think it will be most useful

* to move this `initialize_interpreter()` overload out of the `detail` namespace,
* and the `detail::precheck_interpreter();` call to the very top of that overload, just like for the other overload.

The end result will be that the pair of constructors and the pair of `initialize_interpreter()` overloads each are almost identical, except for the first argument (`bool` vs `PyCconfig`).

It's very little extra work, it looks great already.